core/src/main/java/net/sourceforge/jnlp/config/Defaults.java

```diff
@@ -565,6 +565,18 @@
                     ConfigurationConstants.KEY_JVM_ARGS_WHITELIST,
                     null,
                     null
+            ),
+
+            Setting.createDefault(
+                    ConfigurationConstants.KEY_HTTPCONNECTION_CONNECT_TIMEOUT,
+                    String.valueOf(10000),
+                    null
```

Comment:
Can you use `ValidatorFactory.createRangedIntegerValidator(0, Integer.MAX_VALUE)`

Reply:
done